### PR TITLE
Change error cause for a failure to find a TDL for a zipcode so that …

### PR DESCRIPTION
**Pull request by Ryan-Koch**
…we wind up returning a 404 instead of a 500 from the internalapi handler. Added in a test for this situation for the handler as well.

## Description

Use the ErrFetchNotFound error so that when it bubbles up to the internalapi handler for shipments a 404 is returned instead of a 500.

## Setup

No special steps needed.

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](./docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](./docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](./docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested using `bin/run-prod-migrations`
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/163213149) for this change

### CODE REVIEW

File: pkg/models/shipment.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func (s *Shipment) BeforeSave(tx *pop.Connection) error {
 	// is created/updated.
 	trafficDistributionList, err := s.DetermineTrafficDistributionList(tx)
 	if err != nil {
-		return errors.Wrapf(err, "Could not determine TDL for shipment ID %s for move ID %s", s.ID, s.MoveID)
```
Comment (donaldthai) [inline pkg/models/shipment.go]: I'm curious to know if we should be appending the error we get back from `DetermineTrafficDistributionList()`?

**Comment by jacquelineIO** [comment]
@Ryan-Koch can we also prevent entering in invalid zip codes? What happens when the 500 surfaces? Is the user alerted to the issue? Are we assuming if the 500 is return then the zip is not valid? (I'm not sure that is a good assumption.)

File: pkg/models/shipment.go (hunk 1 above)
Comment (Ryan-Koch) [reply pkg/models/shipment.go]: That's a fair point. Went ahead and added in the error from that call into the message 👍.

**Comment by Ryan-Koch** [comment]
> @Ryan-Koch can we also prevent entering in invalid zip codes? What happens when the 500 surfaces? Is the user alerted to the issue? Are we assuming if the 500 is return then the zip is not valid? (I'm not sure that is a good assumption.)

That's a good question. I'm unsure what the frontend is currently doing with these non-200 responses in this context. I'll take a look and see.

**Comment by Ryan-Koch** [comment]
> @Ryan-Koch can we also prevent entering in invalid zip codes? What happens when the 500 surfaces? Is the user alerted to the issue? Are we assuming if the 500 is return then the zip is not valid? (I'm not sure that is a good assumption.)

@jacquelineIO exploring this led me to find a spot I missed where this was still returning a 500 so that's definitely helpful. As far as how the UI interacts with these things, it looks like we hit some error feedback around spots like choosing your weight for a PPM move. But it doesn't look like we have any validation on postal codes going in. This probably warrants a story. I also had a conversation with @rdhariwal where it was mentioned there's going to be an epic about error handling, which I believe touches on possibly expanding what we're using for these errors the a bubble up to the UI. 

So, action items for me I think are to write up a validation story (if one doesn't exist already) and also to push another commit covering this other 500 returning situation with the zip3.

**Comment by jacquelineIO** [comment]
@Ryan-Koch sounds good! Thank you for the update.

**Pull request merged**